src/activity_handlers.cpp

```diff
@@ -2866,19 +2866,51 @@ void activity_handlers::dig_channel_do_turn( player_activity *act, player *p )
 void activity_handlers::dig_finish( player_activity *act, player *p )
 {
     const tripoint &pos = act->placement;
-
-    if( g->m.ter( pos ) == t_pit_shallow ) {
+    bool grave = g->m.ter( pos ) == t_grave ? true : false;
+    if( g->m.ter( pos ) == t_pit_shallow || grave ) {
         g->m.ter_set( pos, t_pit );
     } else {
         g->m.ter_set( pos, t_pit_shallow );
     }
+    if( grave ) {
+        if( one_in( 10 ) ) {
+            static const std::array<mtype_id, 5> monids = { {
+                    mtype_id( "mon_zombie" ), mtype_id( "mon_zombie_fat" ),
+                    mtype_id( "mon_zombie_rot" ), mtype_id( "mon_skeleton" ),
+                    mtype_id( "mon_zombie_crawler" )
+                }
+            };
+
+            g->summon_mon( random_entry( monids ), pos );
+            g->m.furn_set( pos, f_coffin_o );
+            p->add_msg_if_player( m_bad, _( "Something crawls out of the coffin!" ) );
+        } else {
+            g->m.add_item_or_charges( pos, item( "bone_human" ) );
```

Comment:
Maybe add a several (random number?) bones.

Reply:
Applied, and also added random damage to exhumed clothing items.